subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/transform/ArtifactTransformWithFileInputsIntegrationTest.groovy

```diff
@@ -257,4 +259,121 @@ class ArtifactTransformWithFileInputsIntegrationTest extends AbstractDependencyR
         outputContains("processing c.jar using [tool-a.jar, tool-b.jar]")
         outputContains("result = [b.jar.green, c.jar.green]")
     }
+
+    @Unroll
+    def "can use input path sensitivity #pathSensitivity for parameter object"() {
+        settingsFile << """
+                include 'a', 'b', 'c'
+            """
+        setupBuildWithTransformFileInputs("@PathSensitive(PathSensitivity.$pathSensitivity) @InputFiles")
+        buildFile << """
+            allprojects {
+                ext.inputFiles = files(rootProject.file(project.property('fileName')))
+            }
+            
+            project(':a') {
+                dependencies {
+                    implementation project(':b')
+                    implementation project(':c')
+                }
+            }
+        """
+        def (first, second, third) = files.entrySet()
+
+        when:
+        def firstFilePath = first.key
+        def firstFile = file(firstFilePath)
+        firstFile.text = first.value
+        run(":a:resolve", "-PfileName=${firstFilePath}")
+        then:
+        outputContains("processing b.jar using [$firstFile.name]")
+        outputContains("processing c.jar using [$firstFile.name]")
+        outputContains("result = [b.jar.green, c.jar.green]")
+
+        when:
+        file(second.key).text = second.value
+        run(":a:resolve", "-PfileName=${second.key}")
+        then:
+        outputDoesNotContain("Transform artifact")
+        outputContains("result = [b.jar.green, c.jar.green]")
+
+        when:
+        def thirdFilePath = third.key
+        def thirdFile = file(thirdFilePath)
+        thirdFile.text = third.value
+        run(":a:resolve", "-PfileName=${thirdFilePath}")
+        then:
+        outputContains("processing b.jar using [$thirdFile.name]")
+        outputContains("processing c.jar using [$thirdFile.name]")
+        outputContains("result = [b.jar.green, c.jar.green]")
+
+        where:
+        pathSensitivity | files
+        "NONE"          | [first: 'foo', second: 'foo', third: 'bar']
+        "NAME_ONLY"     | ['first/input': 'foo', 'second/input': 'foo', 'third/input1': 'foo']
+        "RELATIVE"      | ['first/input': 'foo', 'second/input': 'foo', 'third/input1': 'foo']
```

Comment:
Do we need `ABSOLUTE` here, too?

Reply:
Nit: shall we use the actual enum values? Might come in handy when searching for references.

Reply:
Testing `ABSOLUTE` and using the enum values now.